addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js

```diff
@@ -0,0 +1,322 @@
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain('The following are invalid CIDR inputs: invalidCidr');
+      }
+    });
+
+    it('should fail because the updateRequest contains Ipv6 CIDR ranges', async () => {
+      // BUILD
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32', '1234::1234:abcd:ffff:1234:123/64'],
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain(
+          'Currently only IPv4 formatted CIDR inputs are allowed. Please remove: 1234::1234:abcd:ffff:1234:123/64 CIDR ranges from your list',
+        );
+      }
+    });
+
+    it('should fail because the updateRequest contains duplicate protocol-port combinations', async () => {
+      // BUILD
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32'],
+          },
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['456.456.456.456/32'],
+          },
+        ],
+      };
+
+      // OPERATE
+      try {
+        await service.update({}, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(service.boom.is(err, 'badRequest')).toBe(true);
+        expect(err.message).toContain(
+          'The update request contains duplicate protocol-port combinations. Please make sure all ingress rule objects have unique protocol-fromPort-toPort combinations',
+        );
+      }
+    });
+  });
+
+  describe('update', () => {
+    it('should fail because the user is not authorized', async () => {
+      // BUILD
+      const requestContext = {};
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['123.123.123.123/32'],
+          },
+        ],
+      };
+      service.assertAuthorized.mockImplementationOnce(() => {
+        throw new Error('User is not authorized');
+      });
+
+      // OPERATE
+      try {
+        await service.update(requestContext, params);
+        expect.hasAssertions();
+      } catch (err) {
+        expect(err.message).toEqual('User is not authorized');
+      }
+    });
+
+    it('should use the correct CIDR blocks to make revoke and authorize calls', async () => {
```

Comment:
I think there are a few more cases which can be added:

1. Update request and current ingress rules have some overlapping IPs and we have some IPs to revoke and some to register
2. Empty update request (just revoke)
3. Current ingress rules contain the same protocol, fromPort and toPort
4. Current ingress rules don't contain the protocol, fromPort and toPort in the request
5. Update request has IPs but current ingress rules have nothing (just authorize)
6. And adding failure scenarios like any revoke or authorize failure isn't swallowed and exception is thrown

Reply:
Done